packages/ra-core/src/auth/useHandleAuthCallback.spec.tsx

```diff
@@ -49,34 +40,41 @@ const authProvider: AuthProvider = {
         return Promise.resolve();
     },
     getPermissions: () => Promise.reject('not authenticated'),
-    handleCallback: () => Promise.resolve(),
+    handleCallback: () => Promise.resolve({}),
 };
 
-const queryClient = new QueryClient();
-
 describe('useHandleAuthCallback', () => {
     afterEach(() => {
-        redirect.mockClear();
+        jest.clearAllMocks();
     });
 
     it('should redirect to the home route by default when the callback was successfully handled', async () => {
-        const history = createMemoryHistory({ initialEntries: ['/'] });
+        const history = createMemoryHistory({
+            initialEntries: ['/auth-callback'],
+        });
         render(
             <HistoryRouter history={history}>
                 <AuthContext.Provider value={authProvider}>
-                    <QueryClientProvider client={queryClient}>
-                        <TestComponent />
+                    <QueryClientProvider client={new QueryClient()}>
+                        <Routes>
+                            <Route path="/" element={<div>Home</div>} />
+                            <Route path="/test" element={<div>Test</div>} />
+                            <Route
+                                path="/auth-callback"
+                                element={<TestComponent />}
+                            />
+                        </Routes>
```

Comment:
I'm confused, this isn't the same test. Why do you need to put the tested component in routes?

Reply:
Jest mocks were unreliable. The new test actually tests the same behavior as before